### PR TITLE
[Merged by Bors] -  chore(Tactic/NormNum): remove autoImplicit

### PR DESCRIPTION
**Pull request by grunweg**
This is exhaustive

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

~~except for two `norm_num` attributes in `Pow`. Help with those is welcome!~~

### CODE REVIEW

**Review by Ruben-VandeVelde — APPROVED** [review]
maintainer merge

File: Mathlib/Tactic/NormNum/BigOperators.lean [hunk 1]
```diff
@@ -109,7 +109,7 @@ set_option linter.unusedVariables false in
 
 Fails if we cannot determine which of the alternatives apply to the expression.
 -/
-partial def List.proveNilOrCons {α : Q(Type u)} (s : Q(List $α)) :
+partial def List.proveNilOrCons {u : Level} {α : Q(Type u)} (s : Q(List $α)) :
```
Comment (eric-wieser) [inline Mathlib/Tactic/NormNum/BigOperators.lean:112]: Any reason for having this both here, and at the top of the file?

File: Mathlib/Tactic/NormNum/Result.lean [hunk 2]
```diff
@@ -317,15 +320,15 @@ instance : ToMessageData (Result x) where
   | .isRat _ q _ _ proof => m!"isRat {q} ({proof})"
 
 /-- Returns the rational number that is the result of `norm_num` evaluation. -/
-def Result.toRat : Result e → Option Rat
+def Result.toRat {α : Q(Type u)} {e : Q(«$α»)} : Result e → Option Rat
```
Comment (eric-wieser) [inline Mathlib/Tactic/NormNum/Result.lean:323]: Surely the french quotes aren't needed here?

File: Mathlib/Tactic/NormNum/Pow.lean [hunk 3]
```diff
@@ -237,7 +238,7 @@ blocks below were not necessary: we just did it once outside the `match rb with`
 -/
 /-- The `norm_num` extension which identifies expressions of the form `a ^ b`,
 such that `norm_num` successfully recognises both `a` and `b`, with `b : ℤ`. -/
-@[norm_num (_ : α) ^ (_ : ℤ)]
```
Comment (eric-wieser) [inline Mathlib/Tactic/NormNum/Pow.lean:240]: I fixed this for you; the type annotation was just nonsense.

Comment (grunweg) [reply Mathlib/Tactic/NormNum/Pow.lean:240]: Thanks a lot!

File: Mathlib/Tactic/NormNum/BigOperators.lean (hunk 1 above)
Comment (grunweg) [reply Mathlib/Tactic/NormNum/BigOperators.lean:112]: If I remove this, line 142 throws an error
```
application type mismatch
  ?m.5768 v head✝¹ fst✝ β head✝ f xxs this
argument
  this
has type
  ?_pp_uniq.9 =Q ?_pp_uniq.10 : Prop
but is expected to have type
  ?_pp_uniq.12 =Q ?_pp_uniq.13 : PropLean 4

**Comment by grunweg** [comment]
Thanks for the careful comments! I have addressed them all.

**Pull request closed without merging**